src/vs/workbench/contrib/debug/browser/debugEditorActions.ts

```diff
@@ -35,21 +35,25 @@ class ToggleBreakpointAction extends EditorAction {
 	}
 
 	public run(accessor: ServicesAccessor, editor: ICodeEditor): Promise<any> {
-		const debugService = accessor.get(IDebugService);
-
-		const position = editor.getPosition();
-		if (editor.hasModel() && position) {
+		if (editor.hasModel()) {
+			const debugService = accessor.get(IDebugService);
 			const modelUri = editor.getModel().uri;
-			const bps = debugService.getModel().getBreakpoints({ lineNumber: position.lineNumber, uri: modelUri });
-
-			if (bps.length) {
-				return Promise.all(bps.map(bp => debugService.removeBreakpoints(bp.getId())));
-			}
-			if (debugService.getConfigurationManager().canSetBreakpointsIn(editor.getModel())) {
-				return debugService.addBreakpoints(modelUri, [{ lineNumber: position.lineNumber }], 'debugEditorActions.toggleBreakpointAction');
-			}
+			const lineNumbers = new Set(editor._getCursors().getAll()
```

Comment:
Please use `editor.getSelections()` instead of `editor._getCursors`.
That is the proper API (I just double checked)

Reply:
When I did this I thought the Selection object was for highlighted text only, realized later on that was not the case. I made this change. In using the Selection object I am only considering the line the cursor is on. I could have done all lines that are selected, since the original did not do this, I did not either.